trek.js

```diff
@@ -0,0 +1,195 @@
+// solution that's working without closures - see closures attempt below.
+
+const BASE_URL = 'https://trektravel.herokuapp.com/trips'
+
+const reportStatus = (message) => {
+    $('#status').html(message);
+}
+
+const reportApiError = (error) => {
+    console.log("encountered error when posting:", error);
+  
+    let errorHtml = `<p>${error.message}</p><ul>`;
+
+    const fieldProblems = error.response.data.errors;
+  
+    Object.keys(fieldProblems).forEach(field => {
+      const problems = fieldProblems[field];
+      problems.forEach(problem => {
+        errorHtml += `<li><strong>${field}:</strong> ${problem}</li>`;
+      });
+    });
+  
+    errorHtml += '</ul>';
+    reportStatus(errorHtml);
+}
+
+const loadTrips = () => {
+    reportStatus('Loading trips...');
+    
+    axios.get(BASE_URL)
+    .then((response) => {
+        reportStatus(`Successfully loaded ${response.data.length} trips`);
+        $('#trek-list').append(`<h2>Trip List:</h2>`);
+        response.data.forEach((trek) => {
+            $('#trek-list').append(`<li><a href="#" data-trek-id=${trek.id}> ${trek.name} </a></li>`);  
+        });
+        $('#trek-list li').click(showTrip);
+    })
+
+    .catch((error) => {
+        console.log(error);
+    });
+};
+
+const showTrip = (event) => {
+    axios.get(BASE_URL + '/' + $(event.target).data("trek-id"))
+    .then((response) => {
+        $('#trek-details').html(
+        `<h2>Trip details:</h2>
+        <p><strong>Name:</strong> ${response.data.name}</p>
+        <p id="trek-id"><strong>ID:</strong> ${response.data.id}</p>
+        <p><strong>Continent:</strong> ${response.data.continent}</p>
+        <p><strong>Category:</strong> ${response.data.category}</p>
+        <p><strong>Weeks:</strong> ${response.data.weeks}</p>
+        <p><strong>Cost:</strong> $${response.data.cost}</p>
+        <p><strong>About:</strong> ${response.data.about}</p>`);
+        $('#reservation').show();
+        $('#reservation-form').removeClass().addClass(`${response.data.id}`).submit(addRes);
```

Comment:
So every time I click on a trip a **new** event handler gets added to the form.  This means that the `POST` request could get fired off multiple times.